### PR TITLE
[COST-3527/28/29] Enabled tags all together now

### PR DESCRIPTION
**Pull request by adberglund**
## Jira Ticket

[COST-3527](https://issues.redhat.com/browse/COST-3527)
[COST-3528](https://issues.redhat.com/browse/COST-3528)
[COST-3529](https://issues.redhat.com/browse/COST-3529)

## Description

* This change will switch all source types, including OpenShift to using the enabled tag key table as the source of truth for tag key existence. This change includes when checking the validity of a tag key in reports APIs, and the application settings API. 
* Tag/Label summary and value tables will also no exclusively contain information for enabled keys. 

## Testing

1. Checkout Branch
2. Run make create-test-customer and make load-test-customer-data
3. Thrive
4. Run some report queries for disabled tags and find they are not there.

## Notes

...

### CODE REVIEW

File: koku/api/settings/settings.py [hunk 1]
```diff
@@ -112,40 +102,22 @@ def __init__(self, request):
...
-                    if tag_key.enabled:
-                        enabled.append(tag_key.key)
-            else:
-                enabled_tags = tag_keys_kls.objects.all()
-                enabled = [enabled_tag.key for enabled_tag in enabled_tags]
-
-            all_tags_set.update(enabled)
-
-        return all_tags_set, enabled
+            for tag_key in tag_keys_kls.objects.all():
```
Comment (maskarb) [inline koku/api/settings/settings.py:116]: Is it possible to sort via Django here?

**Review by maskarb — DISMISSED** [review]
Looks good to me! Just one question.

File: koku/api/settings/settings.py (hunk 1 above)
Comment (adberglund) [reply koku/api/settings/settings.py:105]: Discussed and switched to using set logic.

**Comment by adberglund** [comment]
Smoke Tests: https://ci.ext.devshift.net/job/project-koku-koku-pr-check/8267/

**Review by maskarb — APPROVED** [review]
lgtm!

**Review by maskarb — APPROVED** [review]

**Review by maskarb — APPROVED** [review]

**Review by maskarb — APPROVED** [review]

**Pull request merged**